splice_machine/src/main/java/com/splicemachine/derby/utils/StatisticsAdmin.java

```diff
@@ -1112,210 +1112,30 @@ public static Iterable displayTableStatistics(ArrayList<StatisticsOperation> col
-                                            try {
-                                                dataDictionary.addColumnStatistics(nextRow, tc);
-                                            } catch (StandardException e) {
-                                                // DB-9890 Skip a column if its statistics object doesn't fit into HBase cell.
-                                                if (e.getCause().getMessage().contains("KeyValue size too large")) {
-                                                    int columnId = nextRow.getColumn(SYSCOLUMNSTATISTICSRowFactory.COLUMNID).getInt();
-                                                    SpliceLogUtils.warn(LOG, "Statistics object of [ConglomID=%d, ColumnID=%d] exceeds max KeyValue size. Try increase hbase.client.keyvalue.maxsize.",
-                                                            nextRow.getColumn(SYSCOLUMNSTATISTICSRowFactory.CONGLOMID).getInt(),
-                                                            columnId);
-                                                    skippedColIds.add(columnId);
-                                                } else {
-                                                    throw e;
-                                                }
-                                            }
-                                            nextRow = input.getNextRowCore();
-                                        }
-                                        fetched = true;
-                                    }
-                                    return nextRow != null;
-                                } catch (Exception e) {
-                                    throw new RuntimeException(e);
-                                }
-                            }
-
-                            @Override
-                            public ExecRow next() {
-                                try {
-                                    fetched = false;
-                                    dataDictionary.addTableStatistics(nextRow, tc);
-                                    Pair<String,String> pair = displayPair.get(nextRow.getColumn(SYSTABLESTATISTICSRowFactory.CONGLOMID).getLong());
-                                    return generateOutputRow(pair.getFirst(),pair.getSecond(),nextRow,skippedColIds);
-                                } catch (Exception e) {
-                                    throw new RuntimeException(e);
-                                }
-                            }
-                        };
-                        return () -> iterator;
-                    } catch (Exception e) {
-                        throw new RuntimeException(e);
-                    }
-                }
-            }).toList();
+            if (mergeStats) {
+                tc.releaseSavePoint("statistics", null);
+            }
+        } catch (Exception e) {
```

Comment:
It would be a good idea to cancel execution of the Futures here if one of them hits an Exception.  For example, move declaration of `results` outside of the try block, and then include:

>             for (Future<Iterable> f : results) {
>                 f.cancel(true);
>             }

Reply:
thank you, fixed